multimedia/VLC/Portfile

```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
-
-                # http://trac.macports.org/ticket/35131
-                ln -s ${prefix}/lib ${destroot}${applications_dir}/VLC.app/Contents/MacOS/lib
-                ln -s ${prefix}/lib/vlc/plugins ${destroot}${applications_dir}/VLC.app/Contents/MacOS/plugins
-                ln -s ${prefix}/lib/vlc/lua ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/lua
-                ln -s ${prefix}/share/locale ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/locale
-
-                # the vlc executable needs to be started with a full path to the app bundle executable
-                # or else the Mac OS X interface will hang beyond even a ^C or ^\ :
-                move ${destroot}${prefix}/bin/vlc ${destroot}${prefix}/bin/vlc.exe
-                system "echo \"#!/bin/sh\nexec \\\"${applications_dir}/VLC.app/Contents/MacOS/VLC\\\" \\\"\\\$\@\\\"\" > ${destroot}${prefix}/bin/vlc"
-                system "chmod 755 ${destroot}${prefix}/bin/vlc"
-            } else {
-                delete ${destroot}${prefix}/bin
-                delete ${destroot}${prefix}/share/applications
-                delete ${destroot}${prefix}/share/man
-            }
+    #
+    # configure isn't autoconf and they do use a dep cache
+    #
+
+    platform darwin 8 {
+        post-patch {
+            reinplace "s:,-compatibility_version,$\(LIBMAJOR\)::" ${worksrcpath}/configure
         }
     }
-    post-activate {
-        if {[variant_isset qt4] || [variant_isset qt5]} {
-            notes-append "The Qt interface modules for VLC are currently dysfunctional. If you encounter bugs with them, please file them with VLC and not MacPorts."
+
+    destroot.target     install-libs install-headers
+
+    post-destroot {
+        file delete -force ${destroot}${prefix}/share/examples
+        # We need to make sure that the linker will use our libraries and not one
+        # from a location like ${prefix}/lib . That's why we use --build-suffix, but
+        # that still requires us to provide pkg-config files with the standard names:
+        foreach pc [glob ${destroot}${FFMPEG_VLC_PREFIX}/lib/pkgconfig/*.pc] {
+            set standardname [strsed ${pc} "s/-VLC.pc/.pc/"]
+            ln -s [file tail ${pc}] ${standardname}
+        }
+        # oblige dependent code to include files from our own renamed header file directories, so it
+        # cannot include mismatching headers by accident (e.g. those from ffmpeg 3.x).
+        foreach dir {libavcodec libavformat libavresample libavutil libpostproc libswresample libswscale} {
+            file rename ${destroot}${FFMPEG_VLC_PREFIX}/include/${dir} ${destroot}${FFMPEG_VLC_PREFIX}/include/${dir}-VLC
+        }
+        foreach dir {libavcodec libavformat libavresample libavutil libpostproc libswresample libswscale} {
+            foreach header [glob -nocomplain ${destroot}${FFMPEG_VLC_PREFIX}/include/*/*.h] {
+                reinplace "s|${dir}/|${dir}-VLC/|g" ${header}
+            }
```

Comment:
it is being configured that way, but without this mess the compiler will pick up the wrong headerfiles.

The mess could be avoided by configuring port:ffmpeg to put its headers elsewhere but that's not an option, and I prefer to let the portfile jump through well-tested hoops rather than obliging the user to deactivate port:ffmpeg in order to build VLC.